[testing][inductor] Allow grad tolerance override

amjames opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118753
* #119809
* __->__ #119844

Introduce `grad_atol` and `grad_rtol` kwargs, default behavior is
preserved by using `atol` and `rtol` values.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

amjames committed:
[testing][inductor] Allow grad tolerance override

Introduce `grad_atol` and `grad_rtol` kwargs, default behavior is
preserved by using `atol` and `rtol` values.

[ghstack-poisoned]

peterbell10 commented on test/inductor/test_torchinductor.py:
```diff
@@ -554,6 +558,8 @@ def check_model_gpu(
         atol=atol,
         rtol=rtol,
+        grad_atol=grad_atol,
+        grad_rtol=grad_rtol,
```
There is another `check_model` call below on line 586.

amjames replied:
oof, missed that one when I split it off, thanks.

amjames committed:
Update on "[testing][inductor] Allow grad tolerance override"


Introduce `grad_atol` and `grad_rtol` kwargs, default behavior is
preserved by using `atol` and `rtol` values.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

amjames commented:
@pytorchbot merge
